exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java

```diff
@@ -1,169 +1,199 @@
+	}
+
+	private void resetPlayerPosition(int number) {
+		playerPositions[number] = 0; //0 TO VARIABLE
+	}
+
+	//AMOUNTOFPLAYERS
+	public int howManyPlayers() {
+		return players.size();
+	}
+
+	public void roll(int roll) {
+		displayMessage(players.get(currentPlayer) + " is the current player");
+		displayMessage("They have rolled a " + roll);
+		if (currentPlayerCanRole(roll)){
+			playerRolls(roll);
+			askQuestion();
+		}
+	}
+
+	private boolean currentPlayerCanRole(int roll) {
+		boolean isRoleOdd = roll % 2 != 0;
+		if (playerInPenaltyBox[currentPlayer] && isRoleOdd) {
+			setIsGettingOutOfPenaltyBox(true);
+			displayMessage(players.get(currentPlayer) + " is getting out of the penalty box");
+		} else if (playerInPenaltyBox[currentPlayer] && !isRoleOdd) {
+			displayMessage(players.get(currentPlayer) + " is not getting out of the penalty box");
+			setIsGettingOutOfPenaltyBox(false);
+			return false;
+		}
+		return true;
+	}
+
+	private void playerRolls(int roll) {
+		playerPositions[currentPlayer] = playerPositions[currentPlayer] + roll;
+		if (playerPositions[currentPlayer] > 11) {
+			playerPositions[currentPlayer] = playerPositions[currentPlayer] - 12; // EXPLANATION?
+		}
+
+		displayMessage(players.get(currentPlayer)
+				+ "'s new location is "
+				+ playerPositions[currentPlayer]);
+		displayMessage("The category is " + currentCategory());
+	}
+
+	private void setIsGettingOutOfPenaltyBox(boolean isGettingOut) {
+		isGettingOutOfPenaltyBox = isGettingOut;
+	}
+
+	private void askQuestion() {
+		switch (currentCategory()){
+			case Pop: displayMessage(popQuestions.removeFirst());
+				break;
+			case Science: displayMessage(scienceQuestions.removeFirst());
+				break;
+			case Sports: displayMessage(sportsQuestions.removeFirst());
+				break;
+			case Rock: displayMessage(rockQuestions.removeFirst());
+				break;
+		}
+	}
+	
+	
+	private QuestionTypes currentCategory() {
+		int playerPosition = playerPositions[currentPlayer];
+		if (isNumberMultipleOf4(playerPosition, 0)) {
+			return QuestionTypes.Pop;
+		}
+		if (isNumberMultipleOf4(playerPosition, 1)) {
+			return QuestionTypes.Science;
+		}
+		if (isNumberMultipleOf4(playerPosition, 2)) {
+			return QuestionTypes.Sports;
+		}
+		return QuestionTypes.Rock;
+	}
+
+	private boolean isNumberMultipleOf4(int number, int offset) {
+		return number % 4 == offset;
+	}
+
+	public boolean wasCorrectlyAnswered() {
+		if (playerInPenaltyBox[currentPlayer]){
+			if (isGettingOutOfPenaltyBox) {
+				displayMessage("Answer was correct!!!!");
+				return isWinnerFromCorrectAnswer();
+			} else {
+				changePlayer();
+				return true;
+			}
+		} else {
+			displayMessage("Answer was corrent!!!!");
+			return isWinnerFromCorrectAnswer();
+		}
+	}
+
+	private boolean isWinnerFromCorrectAnswer() {
+		addCoinToCurrentPlayer();
+		displayMessage(players.get(currentPlayer)
+				+ " now has "
+				+ playerPurses[currentPlayer]
+				+ " Gold Coins.");
+
+		boolean winner = isCurrentPlayerWinner();
```

Comment:
This might be easier to understand in future if the return boolean winner and method determining the value are renamed. The game runner class calling this is for notAWinner which when returning true, continues the game. Reading this method when returning true, it sounds like the player is a winner. Which is opposite logic. 
If this explanation doesn't make sense let me know.

Reply:
Yes it sounds very strange, it makes it out that they are a winner already. I will change this 👍